lib/msal-node/docs/caching.md

```diff
@@ -92,37 +93,18 @@ On confidential client applications that handle users (web apps that sign in use
 * For multi-tenant daemon apps using client credentials grant: **tenantId** (or rather, `<clientId>.<tenantId>`)
 * For web APIs calling other web APIs using OBO: hash of the incoming access token from the user (i.e. the token which will subsequently be exchanged for an OBO token)
 
-For instance, when developing a web app that serve users, implement [ICachePlugin](https://azuread.github.io/microsoft-authentication-library-for-js/ref/interfaces/_azure_msal_common.icacheplugin.html) in a class where user's `homeAccountId` is used for partitioning:
-
-```typescript
-class DistributedCachePlugin implements ICachePlugin {
-
-    // singleton
-    private static instance: CachePlugin;
-
-    private persistenceManager: any; // your implementation of a persistence client
-    private partitionKey: string;
+MSAL Node provides the [DistributedCachePlugin](https://azuread.github.io/microsoft-authentication-library-for-js/ref/classes/_azure_msal_node.distributedcacheplugin.html) class for confidential clients, which implements the [ICachePlugin](https://azuread.github.io/microsoft-authentication-library-for-js/ref/interfaces/_azure_msal_common.icacheplugin.html). An instance of `DistributedCachePlugin` requires:
 
-    private constructor(persistenceManager?: any, partitionKey?: string) {}
+* a client interface, which implements `get` and `set` operations on the persistence server (Redis, MySQL etc.).
+* a partition manager, for reading from and writing to cache with respect to a given **partition key**.
 
-    static getInstance(persistenceManager?: any, partitionKey?: string): CachePlugin {}
-
-    async beforeCacheAccess(cacheContext): Promise<void> {
-        // given a partition key, get the relevant cache portion from the persistence store
-        // deserialize and load the cache into msal's memory
-    }
-    async afterCacheAccess(cacheContext): Promise<void> {
-        // if in-memory cache has changed
-            // serialize and persist the cache using the same partition key
-    }
-}
-```
+Please refer to the [RedisTestApp](../../../samples/msal-node-samples/RedisTestApp/README.md) for an implementation.
 
-It is also a good idea to obtain performance metrics for persistent cache operations, such as **cache hit ratios** from your persistence store. This would give you a reliable picture of how your app scales over time.
+> :bulb: It's also a good idea to obtain performance metrics for persistent cache operations, such as **cache hit ratios** from your persistence store. This would give you a reliable picture of how your app scales over time.
```

Comment:
>performance metrics [](http://example.com/codeflow?start=41&length=19)

are the perf metrics provided by MSAL? In .NET there are metrics provided [doc](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-net-token-cache-serialization?tabs=aspnetcore#monitor-cache-hit-ratios-and-cache-performance) do you have something similar in Node? if so, should probably have a similar section on what to measure and why.

Reply:
This should be possible but at the moment we only have guidance on the [browser side](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/performance.md#measuring-performance) @EmLauber @peterzenz